gui/webdisplay/src/RWebWindowsManager.cxx

```diff
@@ -159,26 +159,42 @@ void RWebWindowsManager::SetUseSessionKey(bool on)
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////
-/// Static method to generate kryptographic key
+/// Static method to generate cryptographic key
+/// Parameter keylen defines length of cryptographic key in bytes
+/// Output string will be hex formatted and includes "-" separator after every 4 bytes
+/// Example for 16 bytes: "fca45856-41bee066-ff74cc96-9154d405"
 
 std::string RWebWindowsManager::GenerateKey(int keylen)
 {
-   // try to randomize??
-   gRandom->SetSeed();
+   std::vector<unsigned char> buf(keylen, 0);
+   if (gSystem->GetCryptoRandom(buf.data(), keylen) != keylen) {
+      R__LOG_ERROR(WebGUILog()) << "Error in gSystem->GetCryptoRandom";
+      gSystem->Abort(11); // this line should abort ROOT application
```

Comment:
Looking into the rest of ROOT, it seems unusual to call `gSystem->Abort()` for a fatal error. How about merging the two lines into
```suggestion
      R__ASSERT(false && "Error in gSystem->GetCryptoRandom");
```

Reply:
I see. I rebased my commit.